### PR TITLE
[bitname/external-dns] Support designate application credentials for external-dns

### PR DESCRIPTION
**Pull request by baurmatt**
### Description of the change

This change allows to configure OpenStack application credentials which are an alternative to username/password.

https://docs.openstack.org/keystone/queens/user/application_credentials.html

### Benefits

Use dedicated application credentials.

### Possible drawbacks

-

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/external-dns/README.md [hunk 1]
```diff
@@ -147,11 +147,13 @@ The command removes all the Kubernetes components associated with the chart and
...
 | `designate.userDomainName`                    | When using the Designate provider, specify the OpenStack user domain name. (optional)                                                                                        | `""`                      |
 | `designate.projectName`                       | When using the Designate provider, specify the OpenStack project name. (optional)                                                                                            | `""`                      |
-| `designate.username`                          | When using the Designate provider, specify the OpenStack authentication username. (optional)                                                                                 | `""`                      |
+| `designate.authType`                          | When using the Designate provider, specify the OpenStack auth type. (optional)                                                                                               | `""`                      |
```
Comment (rafariossaa) [inline bitnami/external-dns/README.md:156]: I guess `authType` should be in a new line and not overriding `username`

File: bitnami/external-dns/values.yaml [hunk 2]
```diff
@@ -324,11 +324,13 @@ designate:
...
+  ## @param designate.applicationCredentialId When using the Designate provider, specify the OpenStack authentication application credential ID. This conflicts with `designate.username`. (optional)
+  ## @param designate.applicationCredentialSecret When using the Designate provider, specify the OpenStack authentication application credential ID. This conflicts with `designate.password`. (optional)
   ## @param designate.authUrl When using the Designate provider, specify the OpenStack authentication Url. (optional)
   ## @param designate.regionName When using the Designate provider, specify the OpenStack region name. (optional)
   ## @param designate.userDomainName When using the Designate provider, specify the OpenStack user domain name. (optional)
   ## @param designate.projectName When using the Designate provider, specify the OpenStack project name. (optional)
-  ## @param designate.username When using the Designate provider, specify the OpenStack authentication username. (optional)
+  ## @param designate.authType When using the Designate provider, specify the OpenStack auth type. (optional)
```
Comment (rafariossaa) [inline bitnami/external-dns/values.yaml:333]: I guess `authType` should be in a new line and not overriding `username`

**Review by rafariossaa — CHANGES_REQUESTED** [review]

**Comment by rafariossaa** [comment]
Could you take a look to my comments ?

**Comment by baurmatt** [comment]
@rafariossaa Thanks for taking a look! :) The diff is only confusing. `username` was present twice in the documentation. See my first commit.

**Comment by rafariossaa** [comment]
Sorry, I missed that.

**Pull request merged**